matscipy/calculators/manybody/calculator.py

```diff
@@ -83,122 +109,135 @@ def __init__(self, atom_type, pair_type, F, G, d1F, d2F, d11F, d22F, d12F, d1G,
 
         self.cutoff = cutoff
 
+        if neighbourhood is None:
+            self.neighbourhood = CutoffNeighbourhood(atom_types=atom_type,
+                                                     pair_types=pair_type)
+        else:
+            self.neighbourhood = neighbourhood
+
+        if not isinstance(self.neighbourhood, Neighbourhood):
+            raise TypeError(
+                f"{self.neighbourhood} is not of type Neighbourhood")
+
     def get_cutoff(self, atoms):
+        """Return a valid cutoff."""
         if np.isscalar(self.cutoff):
             return self.cutoff
 
-        # get internal atom types from atomic numbers
-        elements = set(atoms.numbers)
-
-        # loop over all possible element combinations
-        cutoff = 0
-        for i in elements:
-            ii = self.atom_type(i)
-            for j in elements:
-                jj = self.atom_type(j)
-                p = self.pair_type(ii, jj)
-                cutoff = max(cutoff, self.cutoff[p])
-        return cutoff
+        # Get largest cutoff for all possible pair types
+        atom_types = map(self.atom_type, np.unique(atoms.numbers))
+        pair_types = starmap(self.pair_type, product(atom_types, repeat=2))
```

Comment:
Nice, I've never heard of `starmap` before.

Reply:
I discovered it when writing this code. I used to do an ugly zip pattern:
```python
map(lambda x: f(x[0], x[1]), zip(foo, bar))
# much nicer pattern with starmap
starmap(f, zip(foo, bar))
# even better
map(f, foo, bar)
```